libs/language-server/src/lib/ast/wrappers/value-type/valuetype.ts

```diff
@@ -53,6 +53,14 @@ export interface Valuetype<
    */
   isConvertibleTo(target: Valuetype): boolean;
 
+  /**
+   * Flag if valuetype can be referenced by users.
+   * Examples:
+   *   - Users can (not) reference a valuetype to extend it in a valuetype definition
+   *   - Users can (not) reference a valuetype to parse values in the TableInterpreter block
+   */
+  isReferenceableByUser(): boolean;
```

Comment:
We should document the whole `Valuetype` wrapper construct in the developer docs. 

Among others, there should be a recipe on how to add a new valuetype and whether method `isReferenceableByUser` should be overwritten or not.